### DISCUSSION TITLE
removing modified snapshot file

### DISCUSSION THREAD

**Discussion by rusticek** [moosefs/moosefs#509] [Q&A]
Hi,

on my enviroment im doing daily snapshots of VM images. When i want to do some recover i do again mfsmakesnapshot from actual snapshot and run new VM with it.

```
mfsmakesnapshot images/vm1.qcow2 snapshots/

mfsgeteattr snapshots/vm1.qcow2
snapshots/vm1.qcow2: snapshot

mv snapshots/vm1.qcow2 images/vm1x.qcow2

mfsgeteattr images/vm1x.qcow2
images/vm1x.qcow2: snapshot
```

now i start VM with 'vm1x.qcow2' image, snapshot attribute disapears (i suppose because when VM starts, changes on FS are made)
```
mfsgeteattr images/vm1x.qcow2
images/vm1x.qcow2: -
```

is now safe to do 'rm' without affecting anything else?
`rm images/vm1x.qcow2`

**Comment by chogata** [comment]
It is safe. `rm <somefile>` will always rm just the file pointed at by `<somefile>`, no matter if this particular file is a (part of a) snapshot or not. It does not check the flag. The flag is only there to let the system know which files are a result (unmodified) of a snapshot when using the `mfsrmsnapshot` command. So even if your `images/vm1x.qcow2` still had the flag, you coud `rm` it and it and only it would be deleted, `images/vm1.qcow2` would not be affected.
Snapshot makes a lazy copy of file(s) snapshotted, meaning it does not copy the file data (chunks) immediately, only when needed (when there is a change made either in the original or snapshotted file's content), but from the "regular" filesystem's point of view (operations like `rm`) those are two different files and all operations will act accordingly.
